classes/Kohana/Arr.php

```diff
@@ -127,7 +127,7 @@ public static function path($array, $path, $default = NULL, $delimiter = NULL)
 		{
 			$key = \array_shift($keys);
 
-			if (\ctype_digit($key))
+			if (\ctype_digit((string) $key))
 			{
 				// Make the key an integer
 				$key = (int) $key;
```

Comment:
I'm not actually convinced this serves any purpose, afaik PHP `$array['1']` and `$array[1]` refer to the same thing, so maybe we could safely drop the whole if block/cast.

Reply:
Nah that seems how I'd expect it to be. That will be a killer of a deprecation/BC break if they change that!

Tbf we're lucky that `$arr['foo']` doesn't resolve to `$arr[0]` if the array happens to be an indexed one :)

Reply:
I only realised the other day in something else you could have indexed and associative keys in the same array. The return array of getimagesize is something like `[0=>1920, 1=>1080, 'mime'=>'jpeg']` where 0 is width, 1 is height and mime does what it says on the tin. I guess it's to facilitate list() assuming you just wanted the width and height